packages/contracts-bedrock/scripts/upgrades/holocene/templates/bundle_template.json

```diff
@@ -0,0 +1,96 @@
+{
+  "chainId": $L1_CHAIN_ID,
+  "metadata": {
+    "name": "Holocene Hardfork - Contracts Upgrade",
+    "description": "Upgrades the `SystemConfig.sol`, `MIPS.sol`, `FaultDisputeGame.sol`, and `PermissionedDisputeGame.sol` contracts for Holocene."
+  },
+  "transactions": [
+    {
+      "metadata": {
+        "name": "Upgrade `SystemConfig` proxy",
+        "description": "Upgrades the `SystemConfig` proxy to the new implementation, featuring configurable EIP-1559 parameters."
+      },
+      "to": "$PROXY_ADMIN_ADDR",
+      "value": "0x0",
+      "data": "$TX_1_PAYLOAD",
+      "contractMethod": {
+        "type": "function",
+        "name": "upgrade",
+        "inputs": [
+          {
+            "name": "_proxy",
+            "type": "address"
+          },
+          {
+            "name": "_implementation",
+            "type": "address"
+          }
+        ],
+        "outputs": [],
+        "stateMutability": "nonpayable"
+      },
+      "contractInputsValues": {
+        "_proxy": "$SYSTEM_CONFIG_PROXY_ADDR",
+        "_implementation": "$SYSTEM_CONFIG_IMPL"
+      }
+    },
+    {
+      "metadata": {
+        "name": "Upgrade `CANNON` game type in `DisputeGameFactory`",
+        "description": "Upgrades the `CANNON` game type to the new Holocene deployment, with an updated version of `op-program` as the absolute prestate hash."
+      },
+      "to": "$DISPUTE_GAME_FACTORY_PROXY_ADDR",
+      "value": "0x0",
+      "data": "$TX_2_PAYLOAD",
+      "contractMethod": {
+        "type": "function",
+        "name": "setImplementation",
+        "inputs": [
+          {
+            "name": "_gameType",
+            "type": "uint32"
+          },
+          {
+            "name": "_impl",
+            "type": "address"
+          }
+        ],
+        "outputs": [],
+        "stateMutability": "nonpayable"
+      },
+      "contractInputsValues": {
+        "_gameType": "0",
+        "_impl": "$FDG_IMPL"
+      }
+    },
```

Comment:
This must not be executed for chains only using permissioned games.

Reply:
I've updated the image to allow for conditional usage of the `FaultDisputeGame`. New behavior:
* If `USE_PERMISSIONLESS_FAULT_PROOFS=false`
    * No `FaultDisputeGame` implementation (or an accompanying `DelayedWETH` proxy) is deployed.
    * The upgrade bundle does not include a call to `DisputeGameFactory.setImplementation` for `GameTypes.CANNON`.
* If `USE_PERMISSIONLESS_FAULT_PROOFS=true`
    * A `FaultDisputeGame` and an accompanying `DelayedWETH` proxy is deployed.
    * The upgrade bundle gets an extra call to `DisputeGameFactory.setImplementation` for the `CANNON` game type.

For the sake of not writing a conditional markdown renderer in bash, I've left it so that the markdown files autogenerated as a part of the `superchain-ops` script generation includes the `FaultDisputeGame` notes in the `VALIDATION.md` and `README.md`. The `FaultDisputeGame` address will be reflected as `address(0)` if `USE_PERMISSIONLESS_FAULT_PROOFS=false`, and the consumer can always delete those sections manually.

LMK if this works for you.